ksqldb-execution/src/main/java/io/confluent/ksql/schema/query/QuerySchemas.java

```diff
@@ -118,6 +132,60 @@ public synchronized void trackSerdeOp(
     return Collections.unmodifiableMap(loggerToSchemas);
   }
 
+  private SchemaInfo getKeySchemaInfo(final Map<Boolean, Set<String>> kvLoggerNames,
+                                      final String topicName) {
+    final Set<String> topicNames = kvLoggerNames.getOrDefault(KEY_LOGGER_TYPE, ImmutableSet.of());
+
+    if (topicNames.size() != 1) {
+      final String result = topicNames.size() == 0 ? "Zero" : "Multiple";
+
+      throw new IllegalStateException(result + " key logger names registered for topic."
+          + System.lineSeparator()
+          + "topic: " + topicName
+          + "loggers: " + topicNames
+      );
+    }
+
+    return loggerToSchemas.get(Iterables.getOnlyElement(topicNames));
+  }
+
+  private SchemaInfo getValueSchemaInfo(final Map<Boolean, Set<String>> kvLoggerNames,
+                                        final String topicName) {
+    final Set<String> topicNames = kvLoggerNames.getOrDefault(VALUE_LOGGER_TYPE, ImmutableSet.of());
+
+    // NO_VALUE_SCHEMA_FOUND is returned if no value topic has been detected yet. This happens
+    // when a null value (delete) was processed in the topic as the first record. The internal
+    // serde from the value topic does not call the TrackSerde, which can't track the format type
+    // for the topic.
+    final SchemaInfo valueInfo = loggerToSchemas.getOrDefault(
+        Iterables.getFirst(topicNames, EMPTY_TOPIC_NAME),
+        NO_VALUE_SCHEMA_FOUND);
+
+    if (topicNames.size() > 1) {
+      final Optional<ValueFormat> firstValueFormat = valueInfo.valueFormat();
+
+      // Multiple value topics names may be detected. This happens with left/outer joins. The left
+      // and right streams share a state store which share one changelog topic with different
+      // value formats. The track code, though, is only used for QTT which always executes joins
+      // with same value formats. We only make sure all formats are the same so we return one
+      // SchemaInfo.
+      for (final String name : topicNames) {
+        final Optional<ValueFormat> nextValueFormat = loggerToSchemas.getOrDefault(
+            name, NO_VALUE_SCHEMA_FOUND).valueFormat();
+
+        if (!firstValueFormat.equals(nextValueFormat)) {
+          throw new IllegalStateException("Multiple value logger names registered for topic."
```

Comment:
Assuming my understanding above is correct, can we update this error message to be more clear about what the problem is? If this error is hit, that means that someone wrote a stream-stream join test with differing value formats, and that this is not currently supported by the test framework. It'd also be good to link to the issue that @agavra proposed we file to track a fix for this in the error message, as a way to provide additional context. (Without this clarification, I'd be really confused if I wrote a test I expected to pass and then hit this.)

Reply:
I am still wondering how complex it would be to lift this restriction in this PR. If too complex, I am also happy if we address it in a follow up PR, but given the new stream-stream-join requirement, it might be worth to actually add a stream-stream join QTT test with different value format now...